crypto/schnorr-adaptor-signature/sign_utils.go

```diff
@@ -9,168 +9,250 @@ import (
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 )
 
-const (
-	ModNScalarSize       = 32
-	FieldValSize         = 32
-	JacobianPointSize    = 33
-	AdaptorSignatureSize = JacobianPointSize + ModNScalarSize + 1
-)
-
+// encSign implements the core of the `schnorr_presig_sign` algorithm as defined in the spec.
+// It generates a 65-byte pre-signature using the given private key, nonce,
+// public key, message, and encryption key.
+//
+// The algorithm starts from step 9 since steps 1-8 are handled in EncSign
 func encSign(
 	privKey, nonce *btcec.ModNScalar,
 	pubKey *btcec.PublicKey,
 	m []byte,
 	t *btcec.JacobianPoint,
 ) (*AdaptorSignature, error) {
-	// R' = kG (with blinding in order to prevent timing side channel attacks)
+	// Step 9: Compute R = k0*G
+	// NOTE: later we will negate k0 if needed, and after that k = k0
 	k := *nonce
```

Comment:
why rename the variable here? why not just call the argument `k` to begin with?

Reply:
It's to copy the value of `*nonce` to a new local variable `k`. This is to prevent the function from modifying the value of `nonce` (e.g., when we negate `k`)